libcaf_core/caf/async/future.hpp

```diff
@@ -153,6 +145,63 @@ class future {
     return std::holds_alternative<none_t>(cell_->value);
   }
 
+  auto get() {
+    auto sync = make_counted<detail::beacon>();
+    if (cell_->subscribe(nullptr, action{sync})) {
+      std::ignore = sync->wait();
+    }
+    std::unique_lock guard{cell_->mtx};
+    switch (cell_->value.index()) {
+      default:
+        return res_t{sec::broken_promise};
+      case 1:
+        if constexpr (std::is_void_v<T>)
+          return res_t{};
+        else
+          return res_t{std::get<T>(cell_->value)};
+      case 2:
+        return res_t{std::get<error>(cell_->value)};
+    }
+  }
+
+  auto get(actor_clock::duration_type timeout) {
```

Comment:
```suggestion
  template <class Rep, class Period>
  auto get(std::chrono::duration<Rep, Period> timeout) {
```

And then dispatch to the overload taking an absolute timeout (`now() + timeout`).